code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -276,11 +276,108 @@
 	do_sparks(3, TRUE, src)
 	..()
 
+/obj/item/roombaframe
+	name = "Roomba Frame"
+	desc = "A housing that serves as the base for constructing Roombas."
+	icon = 'icons/obj/janitor.dmi'
+	icon_state = "roombaframe"
+
+/obj/item/roombaframe/attackby(obj/O, mob/user, params)
+	if(isprox(O))
+		to_chat(user, "<span class='notice'>You add [O] to [src].</span>")
+		qdel(O)
+		qdel(src)
+		user.put_in_hands(new /obj/item/bot_assembly/roomba)
+	else
+		..()
+
 /mob/living/simple_animal/bot/cleanbot/medbay
 	name = "Scrubs, MD"
 	bot_core_type = /obj/machinery/bot_core/cleanbot/medbay
 	on = FALSE
 
+//Crossed Wanted Roomba Sprites to be Separate
+/mob/living/simple_animal/bot/cleanbot/roomba
+	name = "\improper Roomba"
+	desc = "A little Roomba, he looks so excited!"
+	icon_state = "roomba0"
+
+/mob/living/simple_animal/bot/cleanbot/roomba/Initialize()
+	. = ..()
+	get_targets()
+	icon_state = "roomba[on]"
+
+	var/datum/job/janitor/J = new/datum/job/janitor
+	access_card.access += J.get_access()
+	prev_access = access_card.access
+
+/mob/living/simple_animal/bot/cleanbot/roomba/turn_on()
+	..()
+	icon_state = "roomba[on]"
+	bot_core.updateUsrDialog()
+
+/mob/living/simple_animal/bot/cleanbot/roomba/turn_off()
+	..()
+	icon_state = "roomba[on]"
+	bot_core.updateUsrDialog()
+	
+/mob/living/simple_animal/bot/cleanbot/roomba/UnarmedAttack(atom/A)
+	if(istype(A, /obj/effect/decal/cleanable))
+		anchored = TRUE
+		icon_state = "roomba-c"
+		visible_message("<span class='notice'>[src] begins to clean up [A].</span>")
+		mode = BOT_CLEANING
+		addtimer(CALLBACK(src, .proc/clean, A), 50)
+	else if(istype(A, /obj/item) || istype(A, /obj/effect/decal/remains))
+		visible_message("<span class='danger'>[src] sprays hydrofluoric acid at [A]!</span>")
+		playsound(src, 'sound/effects/spray2.ogg', 50, 1, -6)
+		A.acid_act(75, 10)
+	else if(istype(A, /mob/living/simple_animal/cockroach) || istype(A, /mob/living/simple_animal/mouse))
+		var/mob/living/simple_animal/M = target
+		if(!M.stat)
+			visible_message("<span class='danger'>[src] smashes [target] with its mop!</span>")
+			M.death()
+		target = null
+
+	else if(emagged == 2) //Emag functions
```

Comment:
please just use the emagged flag

Reply:
These are mobs. They don't have a field for this.